src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/main.js

```diff
@@ -687,14 +702,26 @@ define([
                         this.renderPreview(data);
                     }.bind(this));
 
+                    this.sandbox.on('sulu.preview.changes', this.handleChanges.bind(this));
+
                     this.sandbox.on('sulu.preview.initialize', function(data, restart) {
+                        if (this.previewMode === 'auto') {
+                            // if mode is auto init immediately
+                            this.sandbox.emit('sulu.preview.initialize.force', data, restart);
+                        } else if (this.previewMode === 'on_request') {
+                            // if mode is auto init immediately
+                            this.renderPreviewOnRequest(restart);
+                        }
+                    }.bind(this));
+
+                    this.sandbox.on('sulu.preview.initialize.force', function(data, restart) {
                         data = this.sandbox.util.extend(true, {}, this.data, data);
                         if (!this.preview.initiated) {
                             this.preview.start(data, this.options);
                         } else if (!!restart) {
                             // force reload
-                            this.$preview = null;
                             this.sandbox.dom.remove(this.$preview);
+                            this.$preview = null;
                             this.preview.restart(data, this.options, this.template);
                         }
```

Comment:
And I thought we said once that we just pass a function here, instead of implementing the logic in the listener. Then you also wouldn't have to emit the event a few lines before.